File: bench/h4fixtures/js/main.js
// JavaScript call-form fixture.
class Widget {
    static make() { return new Widget(); }
    bump() {}
}
function free() {}
const obj = { deep: { fn() {} } };
const ns = { Inner: class Inner2 { static go() {} } };

function caller() {
    free();                    // 1. bare call
    const w = Widget.make();   // 2. static member call
    w.bump();                  // 3. method call
    obj.deep.fn();             // 4. 3-level member chain
    ns.Inner.go();             // 5. object-namespace chain
    w?.bump();                 // 6. optional-chain call
    new Widget();              // 7. bare new
    new ns.Inner();            // 8. qualified new
    tag`x`;                    // 9. tagged template
}
function tag(s) { return s; }

File: bench/h4fixtures/ruby/main.rb
# Ruby call-form fixture.
module Util
  def self.tool; end
  module Deep
    def self.deepfn; end
  end
end
class Widget
  def self.make
    Widget.new
  end
  def bump; end
end
def free; end

def caller
  free()               # 1. bare call with parens
  w = Widget.make      # 2. Const.method
  w.bump               # 3. receiver.method
  Util.tool            # 4. Module.method
  Util::Deep.deepfn    # 5. scope-resolution receiver + .method
  Util::tool           # 6. :: as method-call operator
  Widget.new           # 7. constructor call
  free                 # 8. bare no-paren call (parses as identifier)
end
